src/modules/accommodations/components/filter-modal.tsx

```diff
@@ -0,0 +1,290 @@
+      }));
+    }
+  };
+
+  const handleRoomSelection = (rooms: number) => {
+    setFilterParams((prev) => ({
+      ...prev,
+      rooms: rooms,
+    }));
+  };
+
+  const handleAmenitySelection = (amenityKey: keyof AccommodationFilterParams) => {
+    setFilterParams((prev) => ({
+      ...prev,
+      [amenityKey]: !prev[amenityKey],
+    }));
+  };
+
+  const handleApplyFilters = () => {
+    setQueryParams(filterParams);
+    setOpen(false);
+  };
+
+  const handleClearAll = () => {
+    setFilterParams({
+      minPrice: 0,
+      maxPrice: 10000,
+      sortBy: AccommodationListSortBy.Price,
+      sortOrder: SortOrder.Asc,
+    });
+
+    setSearchParams({});
+  };
+
+  return (
+    <Dialog open={open} onClose={() => setOpen(false)}>
+      <DialogContent sx={{ display: 'flex', flexDirection: 'column', maxWidth: '600px' }}>
+        <DialogTitle>
+          <CloseIcon sx={styles.closeIcon} onClick={() => setOpen(false)} />
+          <Typography textAlign="center" fontWeight="600" fontSize="18px">
+            Filter Options
+          </Typography>
+        </DialogTitle>
+
+        <Divider sx={{ marginBottom: '15px', backgroundColor: '#000' }} />
+
+        <Box sx={{ paddingX: '10px' }}>
+          <Typography fontWeight="500" fontSize="16px" sx={{}}>
+            Price Range
+          </Typography>
+          <Typography mb="6px" sx={{ color: '#333' }}>
+            Nightly prices before fees and taxes
+          </Typography>
+          <Slider
+            value={[filterParams.minPrice, filterParams.maxPrice]}
+            onChange={(_, newPrice) => handlePriceRange(newPrice)}
+            valueLabelDisplay="auto"
+            color="secondary"
+            min={0}
+            max={10000}
+          />
+        </Box>
+
+        <Box sx={{ display: 'flex', justifyContent: 'space-between', margin: '8px 10px 0' }}>
+          <TextField
+            sx={{ width: '200px' }}
+            label="Min Price"
+            value={filterParams.minPrice}
+            slotProps={{
+              input: {
+                endAdornment: <Typography>$</Typography>,
+              },
+            }}
+            onChange={(e) => handlePriceChange(Number(e.target.value), true)}
+          />
+          <TextField
+            sx={{ width: '200px' }}
+            label="Max Price"
+            value={filterParams.maxPrice}
+            onChange={(e) => handlePriceChange(Number(e.target.value), false)}
+            slotProps={{
+              input: {
+                endAdornment: <Typography>$</Typography>,
+              },
+            }}
+          />
+        </Box>
+
+        <Divider sx={{ margin: '20px 0 15px 0' }} />
+
+        <Box>
+          <Typography fontWeight="500" mb="8px" fontSize="16px">
+            Rooms
+          </Typography>
+          <Box sx={{ display: 'flex', flexWrap: 'wrap', gap: '2px', justifyContent: 'space-between' }}>
+            {[1, 2, 3, 4, 5].map((room) => (
```

Comment:
This makes things very static. It's better to add another option to select more than 5: `5+`

Reply:
but we haven't this kind of option in backend so I leave it like that.

Reply:
Add it to backend?